### DISCUSSION TITLE
value has been optimized out

### DISCUSSION THREAD

**Discussion by 18222901154** [ggml-org/llama.cpp#4640] [Q&A]
![截图 2023-12-26 18-12-42](https://github.com/ggerganov/llama.cpp/assets/15910582/6b59be8a-1300-4e55-a302-8338620ba6cf)
![截图 2023-12-26 18-13-05](https://github.com/ggerganov/llama.cpp/assets/15910582/7fac6809-aef2-4b9d-8db7-a196ad6a1068)

When gdb debugger in VSCode, value has been optimized out, I want to see the value of the variable, how should it be resolved？

System：6.2.0-39-generic  40~22.04.1-Ubuntu
gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
g++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
GNU gdb (Ubuntu 12.1-0ubuntu1~22.04) 12.1

**Comment by misutoneko** [comment]
Rebuild with -O0 I guess...but there are also some more finegrained solutions:
https://stackoverflow.com/questions/24664865/preventing-the-optimizer-from-optimizing-a-variable-away-in-visual-studio
https://stackoverflow.com/questions/44212568/is-it-possible-to-disable-gcc-g-optimisations-for-specific-sections-of-code

And of course if everything else fails, theres always printf() debugging...
